src/core/namespace.c

```diff
@@ -1878,6 +1879,12 @@ static int apply_one_mount(
 
         assert(what);
 
+        if (m->mode == MOUNT_BIND && m->bind_mount_relabel) {
+                r = label_fix_full(AT_FDCWD, what, mount_entry_unprefixed_path(m), 0);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to fix label of '%s' as '%s': %m", what, mount_entry_path(m));
+        }
```

Comment:
Why not do this in the switch() statement above? Also, does this need to cover MOUNT_BIND_RECURSIVE?